src/Host.cs

```diff
@@ -590,6 +592,24 @@ public Module LoadModule(string path)
             return LoadModule(Path.GetFileNameWithoutExtension(path), File.ReadAllBytes(path));
         }
 
+        /// <summary>
+        /// Loads a <see cref="Module"/> given stream as WebAssembly.
+        /// </summary>
+        /// <param name="moduleName">Name of the module</param>
+        /// <param name="stream">Stream with module data</param>
+        /// <returns>Returns a new <see cref="Module"/>.</returns>
+        public Module LoadModule(string moduleName, Stream moduleStream)
+        {
+            using (StreamReader reader = new StreamReader(moduleStream))
```

Comment:
This `StreamReader` looks unnecessary - you can just call `CopyTo` directly on the provided stream. (Or am I missing something?)

Also, I think it's worth mentioning that `StreamReader` automatically closes the provided stream object when disposed, which is probably not what users would usually expect from a method like this¹. Even if that's the intended behavior, it should probably be explained in the documentation.

¹ Just in case, I checked `System.Drawing.Image.FromStream` and `System.Xml.XmlDocument.Load(Stream)`, both the documentation and the implementation. They don't close the provided stream.

Reply:
Agree with you< I have changed little bit implementation ;)